packages/compass-aggregations/src/components/focus-mode/focus-mode.tsx

```diff
@@ -1,56 +1,151 @@
+  overflow: 'hidden',
+});
+
+const previewAreaStyles = css({
+  width: '25%',
+  paddingTop: spacing[4],
+});
+
+const editorAreaStyles = css({
+  width: '50%',
+  paddingTop: spacing[4],
+  backgroundColor: palette.gray.light3,
+});
+
+type StagePreview = {
+  isLoading: boolean;
+  documents: Document[] | null;
+};
 
 type FocusModeProps = {
   isModalOpen: boolean;
+  stageInput: StagePreview | null;
   stageIndex: number;
-  stage: any;
+  stageOperator: string | null;
+  stageOutput: StagePreview | null;
   onCloseModal: () => void;
 };
 
 export const FocusMode: React.FunctionComponent<FocusModeProps> = ({
   isModalOpen,
+  stageInput,
   stageIndex,
-  stage,
+  stageOperator,
+  stageOutput,
   onCloseModal,
 }) => {
-  if (!isModalOpen) return null;
+  if (!isModalOpen) {
+    return null;
+  }
+
   return (
-    <InfoModal
-      showCloseButton={false}
-      title="Focus Mode"
+    <Modal
+      className={modalStyles}
       size={'large'}
-      onClose={onCloseModal}
+      setOpen={onCloseModal}
       open={isModalOpen}
-      data-testid={`focus-mode-modal-${stageIndex}`}
+      data-testid={"focus-mode-modal"}
     >
-      <Body>In progress feature.</Body>
-      <Body>Focus mode for stage {stage.stageOperator}.</Body>
-    </InfoModal>
+      <ModalBody className={containerStyles}>
+        <div className={headerStyles}>
+          <Body>Focus Mode (in progress feature)</Body>
+        </div>
+        <div className={bodyStyles}>
+          <div className={previewAreaStyles} data-testid="stage-input">
+            <FocusModePreview
+              title='Stage Input'
+              isLoading={stageInput?.isLoading}
+              documents={stageInput?.documents} />
+          </div>
+          <div className={editorAreaStyles} data-testid="stage-editor">
+            <FocusModeEditor
+              index={stageIndex}
+              stageOperator={stageOperator} />
+          </div>
+          <div className={previewAreaStyles} data-testid="stage-output">
+            <FocusModePreview
+              title='Stage Output'
+              isLoading={stageOutput?.isLoading}
+              documents={stageOutput?.documents} /> 
+          </div>
+        </div>
+      </ModalBody>
+    </Modal>
   );
 };
 
 const mapState = ({
   focusMode: { isEnabled, stageIndex },
+  inputDocuments,
   pipelineBuilder: {
     stageEditor: {
       stages,
     }
   }
 }: RootState) => {
-  const stage = stages[stageIndex];
+  const currentStage = stages[stageIndex];
+  const previousStage = stages[stageIndex - 1];
+
+  const stageInput: StagePreview = previousStage
+    ? {
+      isLoading: previousStage.loading,
+      documents: previousStage.previewDocs,
+    } : {
+      isLoading: inputDocuments.isLoading,
+      documents: inputDocuments.documents,
+    };
+
+  const stageOutput: StagePreview | null = currentStage
+    ? {
+      isLoading: currentStage.loading,
+      documents: currentStage.previewDocs,
+    } : null;
```

Comment:
You probably want your preview components to be connected to the store separately from editor, this might be the reason for the rendering lag when selecting a new operator in the focus mode